packages/blocks/src/api/serializer.js

```diff
@@ -41,7 +41,7 @@ export function getBlockDefaultClassName( blockName ) {
 	// Generated HTML classes for blocks follow the `wp-block-{name}` nomenclature.
 	// Blocks provided by WordPress drop the prefixes 'core/' or 'core-' (historically used in 'core-embed/').
 	const className =
-		'wp-block-' + blockName.replace( /\//, '-' ).replace( /^core-/, '' );
+		'wp-block-' + blockName.replace( /\//g, '-' ).replace( /^core-/, '' );
```

Comment:
This was previously only replacing the first '/' with a '-' in the block name. So `namespace/block-name/variation-name` would result in `wp-block-namespace-block-name/variation-name`.

This is different to how its PHP counterpart function (`wp_get_block_default_classname()`) behaves which replaces all occurrences in the string.

Reply:
Echoing what Miguel said -- and given the uncertain future of the `/variation` suffix, which seems to be somewhat controversial -- let's maybe revert the `/g` change and instead append the `-variation` manually to the blockname whenever we need it.

Reply:
59c23577a56